apiserver/facades/agent/deployer/deployer.go

```diff
@@ -156,3 +168,48 @@ func getAllUnits(st *state.State, tag names.Tag) ([]string, error) {
 	}
 	return nil, errors.Errorf("cannot obtain units of machine %q: %v", tag, watch.Err())
 }
+
+// Remove removes every given unit from state, calling EnsureDead
+// first, then Remove. It will fail if the unit is not present.
+func (d *DeployerAPI) Remove(ctx context.Context, args params.Entities) (params.ErrorResults, error) {
+	result := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(args.Entities)),
+	}
+	if len(args.Entities) == 0 {
+		return result, nil
+	}
+
+	for i, entity := range args.Entities {
+		tag, err := names.ParseUnitTag(entity.Tag)
+		if err != nil {
+			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrActionNotAvailable)
```

Comment:
ErrActionNotAvailable？

Reply:
Hmm, not sure how that got there. The original code from the common remove method used ErrPerm.